config/v1/types_infrastructure.go

```diff
@@ -531,9 +531,308 @@ type OvirtPlatformStatus struct {
+	// +kubebuilder:validation:MaxLength=80
+	TagCategory string `json:"tagCategory"`
+}
+
+// VSpherePlatformTopology holds the required and optional vCenter objects - datacenter,
+// computeCluster, networks, datastore and resourcePool - to provision virtual machines.
+type VSpherePlatformTopology struct {
+	// datacenter is the vCenter datacenter in which virtual machines will be located
+	// and defined as the failure domain.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=80
+	Datacenter string `json:"datacenter"`
+
+	// computeCluster as the failure domain
+	// This is required to be a path
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=2048
+	// +optional
+	ComputeCluster string `json:"computeCluster,omitempty"`
+
+	// Hosts has information required for placement of machines on VSphere hosts.
+	// +kubebuilder:validation:Optional
+	// +optional
+	Hosts *VSpherePlatformFailureDomainHosts `json:"hosts,omitempty"`
+
+	// networks is the list of port group network nameswithin this failure domain.
+	// Currently, we only support a single interface per RHCOS virtual machine.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxItems=1
+	// +kubebuilder:validation:MinItems=1
+	// +optional
+	Networks []string `json:"networks,omitempty"`
+
+	// datastore is the name or inventory path of the datastore in which the
+	// virtual machine is created/located.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=2048
+	Datastore string `json:"datastore,omitempty"`
+}
+
+type VSpherePlatformFailureDomainHosts struct {
+	// vmGroupName is the Virtual Machine Group name configured
+	// within a vCenter cluster that is associated with
+	// the corresponding Host Group.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=80
+	VMGroupName string `json:"vmGroupName"`
+
+	// hostGroupName is the Host Group name configured
+	// within a vCenter cluster defining a group
+	// of ESXi hosts.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=80
+	HostGroupName string `json:"hostGroupName"`
+}
+
+// VSpherePlatformVCenterSpec stores the vCenter connection fields.
+// This is used by the vSphere CCM.
+type VSpherePlatformVCenterSpec struct {
+
+	// server is the fully-qualified domain name or the IP address of the vCenter server.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=255
+	Server string `json:"server"`
+
+	// port is the TCP port that will be used to communicate to
+	// the vCenter endpoint. This is typically unchanged from
+	// the default of HTTPS TCP/443.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=32767
```

Comment:
We can't use beyond 32767 since those are dynamic
https://serverfault.com/questions/103626/what-is-the-maximum-port-number

```
sso-user@vcs8e-vc [ ~ ]$ cat /proc/sys/net/ipv4/ip_local_port_range
32768   60999
sso-user@vcs8e-vc [ ~ ]$

sso-user@vcs8e-vc [ ~ ]$ sysctl net.ipv4.ip_local_port_range
net.ipv4.ip_local_port_range = 32768    60999
sso-user@vcs8e-vc [ ~ ]$
```